File: wheystore/wheystore.Web/Modules/ServerTypes/SanPhamDB/ShippersForm.ts
import { initFormType, PrefixedContext, StringEditor } from "@serenity-is/corelib";

export interface ShippersForm {
    MaShipperCode: StringEditor;
    TenCongTy: StringEditor;
    TenShipper: StringEditor;
    SoDienThoai: StringEditor;
}

export class ShippersForm extends PrefixedContext {
    static readonly formKey = 'SanPhamDB.Shippers';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!ShippersForm.init) {
            ShippersForm.init = true;

            var w0 = StringEditor;

            initFormType(ShippersForm, [
                'MaShipperCode', w0,
                'TenCongTy', w0,
                'TenShipper', w0,
                'SoDienThoai', w0
            ]);
        }
    }
}

File: wheystore/wheystore.Web/Modules/SanPhamDB/Shippers/ShippersForm.cs
namespace wheystore.SanPhamDB.Forms;

[FormScript("SanPhamDB.Shippers")]
[BasedOnRow(typeof(ShippersRow), CheckNames = true)]
public class ShippersForm
{
    public string MaShipperCode { get; set; }
    public string TenCongTy { get; set; }
    public string TenShipper { get; set; }
    public string SoDienThoai { get; set; }
}